lib/pbench/server/api/auth.py

```diff
@@ -76,6 +76,20 @@ def get_secret_key(self):
         except Exception as e:
             Auth.logger.exception(f"{__name__}: ERROR: {e.__traceback__}")
 
+    def verify_user(self, username):
+        """
+        Check if the provided username belongs to the current user by
+        querying the Usermodel with the current user
+        :param username:
+        :return: User (UserModel instance), verified status (boolean)
+        """
+        user = self.token_auth.current_user()
+        # check if the current username matches with the one provided
+        verified = user is not None and user.username == username
+        Auth.logger.warning("verified status of user '{}' is '{}'", username, verified)
+
+        return user, verified
```

Comment:
And ... as we've started to veer in this direction... do we want a named tuple here to improve documentation?